dailyjs/shared/contexts/TracksProvider.js

```diff
@@ -49,35 +215,32 @@ export const TracksProvider = ({ children }) => {
         participant,
       });
     };
-    const handleParticipantUpdated = ({ participant }) => {
-      dispatch({
-        type: UPDATE_TRACKS,
-        participant,
-      });
-    };
 
     const joinedSubscriptionQueue = [];
 
     const handleParticipantJoined = ({ participant }) => {
       joinedSubscriptionQueue.push(participant.session_id);
     };
 
-    const batchInterval = setInterval(() => {
+    const joinBatchInterval = setInterval(() => {
       if (!joinedSubscriptionQueue.length) return;
       const ids = joinedSubscriptionQueue.splice(0);
-      const participants = callObject.participants();
+      const callParticipants = callObject.participants();
       const updates = ids.reduce((o, id) => {
-        const { subscribed } = participants?.[id]?.tracks?.audio;
+        const { subscribed } = callParticipants?.[id]?.tracks?.audio;
+        const result = { ...o[id] };
         if (!subscribed) {
-          o[id] = {
-            setSubscribedTracks: {
-              audio: true,
-              screenAudio: true,
-              screenVideo: true,
-            },
+          result.setSubscribedTracks = {
+            audio: true,
+            screenAudio: true,
+            screenVideo: true,
           };
         }
-        return o;
+
+        if (rtcpeers?.getCurrentType?.() === 'peer-to-peer') {
+          result.setSubscribedTracks.video = true;
```

Comment:
We need an additional check somewhere around here, I get `TypeError: result.setSubscribedTracks is undefined` -- although not reliably!